src/nvim/os/pty_process_unix.c

```diff
@@ -36,13 +36,27 @@
 # include "os/pty_process_unix.c.generated.h"
 #endif
 
+/// termios saved at startup (for TUI) or initialized by pty_process_spawn().
+static struct termios termios_default;
+
+/// Saves the termios properties associated with `tty_fd`.
+///
+/// @param tty_fd   TTY file descriptor, or -1 if not in a terminal.
+void pty_process_save_termios(int tty_fd)
+  FUNC_ATTR_NONNULL_ALL
+{
+  if (tty_fd == -1 || tcgetattr(tty_fd, &termios_default) != 0) {
+    return;
```

Comment:
Some code cleanup is needed, but I would first like comments on whether the idea is sound.